src/NodaTime/TimeInterval.cs

```diff
@@ -0,0 +1,284 @@
+        private readonly LocalTime? end;
+        
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TimeInterval"/> struct.
+        /// The TimeInterval includes the <paramref name="start"/> LocalTime and excludes the
+        /// <paramref name="end"/> LocalTime. The end may equal the start (resulting in an empty TimeInterval).
+        /// </summary>
+        /// <param name="start">The start <see cref="LocalTime"/>.</param>
+        /// <param name="end">The end <see cref="LocalTime"/>.</param>
+        public TimeInterval(LocalTime start, LocalTime end)
+        {
+            this.start = start;
+            this.end = end;
+            IsEndInNextDay = start > end;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TimeInterval"/> struct from two nullable <see cref="LocalTime"/>
+        /// values.
+        /// </summary>
+        /// <remarks>
+        /// If the start is null, the TimeInterval is deemed to stretch to the start of time. If the end is null,
+        /// the TimeInterval is deemed to stretch to the end of time.
+        /// </remarks>
+        /// <param name="start">The start <see cref="LocalTime"/> or null.</param>
+        /// <param name="end">The end <see cref="LocalTime"/> or null.</param>
+        public TimeInterval(LocalTime? start, LocalTime? end)
+        {
+            this.start = start;
+            this.end = end;
+            if (start.HasValue && end.HasValue)
+                IsEndInNextDay = start > end;
+            else
+                IsEndInNextDay = false;
+        }
+
+        /// <summary>
+        /// Gets the start LocalTime - the inclusive lower bound of the TimeInterval.
+        /// </summary>
+        /// <value>The start <see cref="LocalTime"/>.</value>
+        /// <exception cref="InvalidOperationException">The TimeInterval extends to the start of time.</exception>
+        /// <seealso cref="HasStart"/>
+        public LocalTime Start
+        {
+            get
+            {
+                Preconditions.CheckState(start.HasValue, "TimeInterval extends to start of time");
+                return start.GetValueOrDefault();
+            }
+        }
+
+        /// <summary>
+        /// Returns <c>true</c> if this TimeInterval has a fixed start point, or <c>false</c> if it
+        /// extends to the start of time.
+        /// </summary>
+        /// <value><c>true</c> if this TimeInterval has a fixed start point, or <c>false</c> if it
+        /// extends to the start of time.</value>
+        public bool HasStart => start != null;
+
+        /// <summary>
+        /// Gets the end LocalTime - the exclusive upper bound of the TimeInterval.
+        /// </summary>
+        /// <value>The end <see cref="LocalTime"/>.</value>
+        /// <exception cref="InvalidOperationException">The TimeInterval extends to the end of time.</exception>
+        /// <seealso cref="HasEnd"/>
+        public LocalTime End
```

Comment:
It's at least worth *considering* just making this nullable. I couldn't do that in Interval because we only made use of nullable instants after 1.0 had shipped.
I'm not sure of the right thing to do in this particular case.

Reply:
Leaved as is. We should think more about it.